app/controllers/api_controller/parameters.rb

```diff
@@ -111,15 +111,17 @@ def sort_params(klass)
       return [] if params['sort_by'].blank?
 
       orders = String(params['sort_order']).split(",")
+      options = String(params['sort_options']).split(",")
       params['sort_by'].split(",").zip(orders).collect do |attr, order|
         raise BadRequestError,
               "#{attr} is not a valid attribute for #{klass.name}" if !klass.respond_to?(attr) && attr != "id"
-        sort_directive(attr, order)
+        sort_directive(attr, order, options)
       end.compact
     end
 
-    def sort_directive(attr, order)
+    def sort_directive(attr, order, options)
       sort_item = attr
+      sort_item = "LOWER(#{sort_item})" if options.map(&:downcase).include?("nocase")
```

Comment:
`nocase` feels weird as an option name.  Was googling around and came across `ignore_case`, which seems nice.  Perhaps also `case_insensitive` or just `insensitive`